Update Startup.cs

B-Virtual opened the pull request:
Changed the order of the appBuilder initialization because the current tenant middleware will not trigger if an API route is executed. No tenant is present in the DB context and the queries do not work correctly. Using .UseCurrentTenant() before .UseRouting() fixes this.

fretje commented:
Hmm... according to the [docs](https://docs.microsoft.com/en-us/aspnet/core/migration/22-to-30?view=aspnetcore-3.0&tabs=visual-studio#migrate-startupconfigure), and also explained [here](https://andrewlock.net/inserting-middleware-between-userouting-and-useendpoints-as-a-library-author-part-1/), the UseRouting should always come before UseAuthentication and UseAuthorization...

B-Virtual commented:
Indeed, that is correct.

The issue I had was that the CurrentTenantMiddleware was only invoked for pages like Swagger and the Swagger index page but not for the API routes.

I changed it back, cleaned my solution and restarted VS 2022 and apparently now the middleware is triggered on API calls too. I spend some time figuring out why the tenant was not present and pinned it down to the middleware but it was a ghost apparently. Sorry for that, I'll change it back.
